R/framework_ebp.R

```diff
@@ -45,9 +48,19 @@ framework_ebp <- function(fixed, pop_data, pop_domains, smp_data, smp_domains,
                                     levels = levels_tmp)
   pop_domains_vec <- pop_data[[pop_domains]]
 
-
   smp_data[[smp_domains]] <- factor(smp_data[[smp_domains]],
                                     levels = levels_tmp)
+
+  if(is.null(aggregate_to) != TRUE){
+    levels_tmp <- unique(pop_data[[aggregate_to]])
+    pop_data[[aggregate_to]] <- factor(pop_data[[aggregate_to]],
+                                       levels = levels_tmp)
+    aggregate_to_vec <- pop_data[[aggregate_to]]
+  }else{
+    aggregate_to_vec <- NULL
+  }
+
+
   rm(levels_tmp)
   smp_data <- smp_data[order(smp_data[[smp_domains]]), ]
```

Comment:
Have you tested the code with other data than the example data? We had huge ordering issues when the domains in sample and census data are not ordered the same and we just fixed it. There seems to be more ordering now and level defining so we should be sure that we do not get a similar error again.

Reply:
Habe ich geprüft mit einem "ungeordneten" synthetischen Datensatz.